### ISSUE TITLE
Lint to detect empty comments within docs ("`//` within `///`")

### ISSUE THREAD

**Issue by ojeda** [rust-lang/rust-clippy#15245]
### What it does

Sometimes a slash may be missed within (or adjacent to) a larger comment, which can affect rendering of the docs.

The lint should lint when there is an empty comment (`//`) within docs (`///`, `//!`). Note that it should not lint in cases where there is actually something written, e.g. a `// TODO:` in the middle of docs. It should not lint either if there is any non-empty comment line in the same group of comment lines.

This happened in a patch to the Linux kernel: https://lore.kernel.org/rust-for-linux/CANiq72nh71s9to5v1KHJWN79bEFv97zN6jcGJyEQkaJZ5UuJfg@mail.gmail.com/.

Cc: `rustdoc` @GuillaumeGomez since this is one of those documentation-related ones.

### Example

```rust
/// ...
//
/// ...
```

could be written as:

```rust
/// ...
///
/// ...
```

Similarly, empty comments at the beginning or end of docs should also be linted:

```rust
//
/// ...
```

```rust
/// ...
//
```

The lint should not trigger if there is content in the comment, however:

```rust
/// ...
// ...
/// ...
```

The lint should not trigger if there is content in another line in the same group of consecutive `//` lines:

```rust
/// ...
//
// ...
```

### On a possible extension that lints on non-empty comments

From https://github.com/rust-lang/rust-clippy/pull/15260#issuecomment-3065912533:

> So I think there are these cases:
> 
>   - Surrounding, empty in all lines.
>   - Surrounding, with some content in at least one line.
>   - Adjacent, empty in all lines.
>   - Adjacent, with some content in at least one line.
>
> Of those, 1 and 3 seem both pretty safe to enable and that was my original intention with the issue.
>
> 4 is not something we would want at least for the Linux kernel, since we expect to write those (and we have 6 cases already).
>
> And finally 3 could perhaps be interesting for some projects -- in the Linux kernel we don't have any case yet (if I grepped correctly), but when I wrote our coding guidelines about comments I expected it to be possible (https://docs.kernel.org/rust/coding-guidelines.html#comments):
>
> ```rust
> /// # Examples
> ///
> // TODO: Find a better example.
> /// ```
> /// let foo = f(42);
> /// ```
> ```

**Comment by lolbinarycat** [comment]
what should it do in this case?

```rs
/// ...
// .
//
/// ...
```

**Comment by ojeda** [comment]
For that case, I think the safe option is to not trigger -- that would be in the "if there is content in another line in the same group of consecutive `//` lines" case. The idea is that someone may be doing that on purpose, e.g. to add some visual separation with between docs and a `// TODO:`. But I may be misunderstanding what you mean by the example.

If you mean that the second `//` line doesn't look great, then I agree, but I think that could be a different lint, i.e. one about empty comment/doc lines at the start or end (which I guess could be made configurable whether to lint on your example or not, i.e. whether to lint if the empty lines are within mixed docs/comments).

Relatedly, sometimes empty `//` lines at the start/end could be intentional, e.g. I had to use an empty `//` to separate groups of `#[feature(...)]`s due to `rustfmt`: https://github.com/Rust-for-Linux/linux/blob/rust-next/rust/kernel/lib.rs#L25-L32. But, yeah, I don't think it is common.

Thanks for taking a look!

**Comment by GuillaumeGomez** [comment]
I will try to take a look in the next days.

**Comment by ojeda** [comment]
Thanks @GuillaumeGomez, that was quick! -- I left a comment there (https://github.com/rust-lang/rust-clippy/pull/15260#issuecomment-3065912533), since I think that lint is stronger than what I tried to cover here.

I also edited the OP to mention that they could be at the start/end too and the summary of possible cases.
